src/test/regress/input/hcatalog_lookup.source

```diff
@@ -138,16 +138,51 @@ alter table test_schema.p exchange partition p1 with table hcatalog.test_schema.
 select pg_catalog.pg_database_size('hcatalog');
 select pg_catalog.pg_database_size(6120);
 
+--positive test: should be able to create table named "hcatalog"
+CREATE TABLE hcatalog(a int);
+
+--negative test: cannot create database named "hcatalog"
+CREATE DATABASE hcatalog;
+
+--allow renaming schemas and databases
```

Comment:
@GodenYao enabling this GUC allows rename both schemas and databases in general, so it's correct. By default HAWQ doesn't support renaming databases and schemas, only from pg_dump utility, added this in doc.